core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2508,6 +2508,27 @@ private[spark] object Utils extends Logging {
   }
 }
 
+private[util] object CallerContext extends Logging {
+  val callerContextSupported: Boolean = {
+    val conf = SparkHadoopUtil.get.conf
+    conf.getBoolean("hadoop.caller.context.enabled", false) && {
```

Comment:
Although this construct looks funny to me, a more conventional structure would just be a little longer. I'm OK with it.